src/particles/spacecharge/GatherAndPush.cpp

```diff
@@ -76,32 +81,67 @@ namespace impactx::particles::spacecharge
                 amrex::ParticleReal const push_consts = dt * charge * inv_gamma2 / pz_ref_SI;
 
                 // gather to each particle and push momentum
-                amrex::ParallelFor(np, [=] AMREX_GPU_DEVICE (int i) {
-                    // access SoA Real data
-                    amrex::ParticleReal & AMREX_RESTRICT x = part_x[i];
-                    amrex::ParticleReal & AMREX_RESTRICT y = part_y[i];
-                    amrex::ParticleReal & AMREX_RESTRICT z = part_z[i];
-                    amrex::ParticleReal & AMREX_RESTRICT px = part_px[i];
-                    amrex::ParticleReal & AMREX_RESTRICT py = part_py[i];
-                    amrex::ParticleReal & AMREX_RESTRICT pz = part_pz[i];
-
-                    // force gather
-                    amrex::GpuArray<amrex::Real, 3> const field_interp =
-                        ablastr::particles::doGatherVectorFieldNodal (
-                            x, y, z,
-                            scf_arr_x, scf_arr_y, scf_arr_z,
-                            invdr,
-                            prob_lo);
-
-                    // push momentum
-                    px += field_interp[0] * push_consts;
-                    py += field_interp[1] * push_consts;
-                    pz += field_interp[2] * push_consts;
-
-                    // push position is done in the lattice elements
-                });
-
-
+                if (space_charge == SpaceChargeAlgo::True_2D) {
+                    // flatten 3rd dimension
+                    auto prob_lo_2D = gm.ProbLoArray();
+                    prob_lo_2D[2] = 0.0_rt;
+
+                    // TODO: add in z-dependent scaling by current?
+
+                    amrex::ParallelFor(np, [=] AMREX_GPU_DEVICE (int i) {
+                        // access SoA Real data
+                        amrex::ParticleReal & AMREX_RESTRICT x = part_x[i];
+                        amrex::ParticleReal & AMREX_RESTRICT y = part_y[i];
+                        amrex::ParticleReal z = 0.0_prt;  // flatten 3rd dimension
+                        amrex::ParticleReal & AMREX_RESTRICT px = part_px[i];
+                        amrex::ParticleReal & AMREX_RESTRICT py = part_py[i];
+                        amrex::ParticleReal & AMREX_RESTRICT pz = part_pz[i];
+
+                        // force gather
+                        amrex::GpuArray<amrex::Real, 3> const field_interp =
+                            ablastr::particles::doGatherVectorFieldNodal<2>(
+                                x, y, z,
+                                scf_arr_x, scf_arr_y, scf_arr_z,
+                                invdr,
+                                prob_lo_2D
+                            );
+
+                        // push momentum
+                        px += field_interp[0] * push_consts * beta_gamma * dr[2] / (c0_SI);
+                        py += field_interp[1] * push_consts * beta_gamma * dr[2] / (c0_SI);  //this should be field_interp[1]
```

Comment:
The constants appearing in px and py are currently in draft form only, and will continue to be updated.